internal-services/catalog/file-updates-process-file-updates-task.yaml

```diff
@@ -0,0 +1,126 @@
+            E.g. '[{"path":"file1.yaml","replacements":[{"key":".yamlkey1,","replacement":"|regex|replace|"}]}]'
+  results:
+      - name: fileUpdatesInfo
+        description: fileUpdates detailed information
+      - name: fileUpdatesState
+        description: fileUpdates state
+  steps:
+      - name: perform-updates
+        image: quay.io/hacbs-release/release-utils@sha256:b850b1a4e93d16050eefdb967b7875c17d7af4e86ecb0b268674dc6c3e8d90b1
+        env:
+          - name: GITLAB_HOST
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: gitlab_host
+          - name: ACCESS_TOKEN
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: gitlab_access_token
+          - name: GIT_AUTHOR_NAME
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: git_author_name
+          - name: GIT_AUTHOR_EMAIL
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: git_author_email
+        script: |
+            #!/usr/bin/env sh
+            set -eo pipefail
+
+            # loading git and gitlab functions
+            . /home/utils/gitlab-functions
+            . /home/utils/git-functions
+
+            gitlab_init
+
+            # saves the params.paths json to a file
+            updatePathsTmpfile="/tmp/updatePaths.json"
+            cat > "${updatePathsTmpfile}" << JSON
+            $(params.paths)
+            JSON
+
+            REPO="$(params.repo)"
+            REVISION="$(params.ref)"
+
+            echo -e "=== UPDATING ${REPO} ON BRANCH ${REVISION} ===\n"
+
+            cd /tmp/
+            git_clone_and_checkout --repository "${REPO}" --revision "${REVISION}"
+
+            # getting the files that have replacements
+            PATHS_LENGTH="$(jq '. | length' ${updatePathsTmpfile})"
+            for (( PATH_INDEX=0; PATH_INDEX < $PATHS_LENGTH; PATH_INDEX++ )); do
+                targetFile="$(jq -cr ".[${PATH_INDEX}].path" ${updatePathsTmpfile})"
+
+                # we need to know how many empty newlines and `---` the file has before the actual yaml data starts excluding comments
+                blankLinesBeforeYaml="$(awk '/[[:alpha:]]+/{ if(! match($0, "^#")) { print NR-1; exit } }' ${targetFile})"
+
+                # getting the replacements for the file
+                REPLACEMENTS_LENGTH="$(jq -cr ".[${PATH_INDEX}].replacements | length" ${updatePathsTmpfile})"
+                for (( REPLACEMENT_INDEX=0; REPLACEMENT_INDEX < $REPLACEMENTS_LENGTH; REPLACEMENT_INDEX++ )); do
+                    key="$(jq -cr ".[${PATH_INDEX}].replacements[${REPLACEMENT_INDEX}].key" ${updatePathsTmpfile})"
+                    replacement="$(jq -cr ".[${PATH_INDEX}].replacements[${REPLACEMENT_INDEX}].replacement" ${updatePathsTmpfile})"
```

Comment:
I wonder if we should at least somehow validate that the `replacement` is a valid string for `sed` to consume. Maybe we can add something later.

Reply:
I agree. Also on doing this later. I can add a ticket for this.

Reply:
ticket create and added to the backlog.